risc0/zkvm/src/host/recursion/receipt.rs

```diff
@@ -97,8 +101,22 @@ impl SuccinctReceipt {
             seal_claim.push_back(elem.as_u32())
         }
 
-        // TODO: Read root hash
-        seal_claim.drain(0..16);
+        // NOTE: read_sha_halfs is a misnomer here, because the digest is Poseidon2, but we don't
+        // attempt to do anything with the hash that would make this a problem.
+        let control_root =
+            read_sha_halfs(&mut seal_claim).map_err(|_| VerificationError::ReceiptFormatError)?;
```

Comment:
Is the Poseidon2 hash used actually encoded in the globals as halfs?  I thought that Poseidon hashes get written to the globals as 16 elements with every other one being zero or something weird like that.

Reply:
@jbruestle, I've fixed this in https://github.com/risc0/risc0/pull/1720/commits/7ccba07c7063208d9ff35736df5a871c06430f92

I thought about creating a new "decode poseidon" function, but I'm not sure it would actually be used anywhere else, and it's kind of an odd encoding. I decided to forgo the function, include the logic inline, and comment on what it is doing.